ast/src/analyzed/mod.rs

```diff
@@ -163,14 +163,16 @@ impl<T> Analyzed<T> {
                 poly.id = replacements[&poly_id].id;
             }
         });
-        self.pre_visit_expressions_mut(&mut |expr| {
+        let visitor = &mut |expr: &mut Expression<_>| {
             if let Expression::Reference(Reference::Poly(poly)) = expr {
                 poly.poly_id = poly.poly_id.map(|poly_id| {
                     assert!(!to_remove.contains(&poly_id));
                     replacements[&poly_id]
                 });
             }
-        });
+        };
+        self.post_visit_expressions_in_definitions_mut(visitor);
```

Comment:
Changes from pre to post visit, but the difference between doesn't matter here, right?

Reply:
Yes, it doesn't matter - I didn't want to implement both versions if only one is needed - we could also implement a specialized version that only calls the callback on references.